### PR TITLE
Add lint.py script

### LINKED ISSUE #37: Offline Testing: document tests and provide script
At the moment, we only test with isort and flake8 in the CI. We should document (and maybe simplify trough a script) the procedure to execute tests/lint/... offline, and encourage developers to do so. This avoids the hassle of doing "one-character-correction-commits".

The best place is probably over here: https://chaosinventory.readthedocs.io/projects/core/en/latest/dev/setup/index.html#working-on-and-with-the-code

We should also provide a script, which allows for local testing of "merge-readiness".

### LINKED ISSUE #53: Implement lint script as python script
Would it be possible to implement that Script as standalone Python Script (but not bash?)
That would allow for simple execution also on non-standard systems like Windows.

_Originally posted by @margau in https://github.com/chaosinventory/chaosinventory/issues/44#issuecomment-791974387_

Ref #44

### LINKED ISSUE #58: Cleanup flake8 and isort
At the moment, we have two locations where we are calling flake8 and isort in an automated fashion:
The CI pipeline, and the lint.py-script. I'd suggest streamlining that and add an "--ci"-Mode to the lint.py-script, making the lint.py-script the single source of truth for all codestyle-related (semi-)automated calls.
Also, the output and general interface of lint.py could be improved after the basics are merged with #44.

### PR DESCRIPTION
**Pull request by e1mo**
Close #37

- [x] Document our tools
- [x] Add lint script
    - [x] ~~Simple shell script~~
    - [x] OS Independent python script (@margau ) close (close #53)
    - [x] Migrate GitHub Actions to this script (close #58)

### CODE REVIEW

**Comment by margau** [comment]
Would it be possible to implement that Script as standalone Python Script (but not bash?)
That would allow for simple execution also on non-standard systems like Windows.

**Comment by e1mo** [comment]
> Would it be possible to implement that Script as standalone Python Script (but not bash?)
> That would allow for simple execution also on non-standard systems like Windows.

In our matrix room we decided that this could be done after this is merged, I've created #53

File: scripts/lint.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+all_succeeded = True
+
+# Analysis 1: isort
+# Run as subprocess for maximum compatibility
+
+print("Running isort")
+
+isort_subprocess_args = ["isort", "--color", "-q", "."]
+if not args.fix:
```
Comment (e1mo) [inline scripts/lint.py:26]: I personally would be in favor of separating checking and fixing.

**Comment by e1mo** [comment]
Regarding the changes in a65f6c3:

All OK:

```
Testing the code under /home/e1mo/Documents/Projekte/chaosinventory/chaosinventory/src

Running isort...
passed

Running flake8...
passed

Summary:	passed
isort:		passed
flake8:		passed
```

Flake fails:

```
Testing the code under /home/e1mo/Documents/Projekte/chaosinventory/chaosinventory/src

Running isort...
passed

Running flake8...
./manage.py:13:9: E303 too many blank lines (2)
        from django.core.management import execute_from_command_line
        ^
./chaosinventory/__main__.py:13:9: E303 too many blank lines (2)
        from django.core.management import execute_from_command_line
        ^
failed

Summary:	failed
isort:		passed
flake8:		failed
```

Isort fails:

```
Testing the code under /home/e1mo/Documents/Projekte/chaosinventory/chaosinventory/src

Running isort...
ERROR: /home/e1mo/Documents/Projekte/chaosinventory/chaosinventory/src/manage.py Imports are incorrectly sorted and/or formatted.
ERROR: /home/e1mo/Documents/Projekte/chaosinventory/chaosinventory/src/manage.py Imports are incorrectly sorted and/or formatted.
failed

Running flake8...
passed

Summary:	failed
isort:		failed
flake8:		passed
```

Colored output looks like this, in this case both failed:

image

**Review by margau — APPROVED** [review]

**Review by em0lar — APPROVED** [review]

**Review by margau — APPROVED** [review]

**Pull request merged**